### PR TITLE
Classification toggle

### LINKED ISSUE #2306: Toggle-able classification color palettes feature
**Is your feature request related to a problem? Please describe.**
For our "classification" products (e.g., [Land Cover Type](https://worldview.earthdata.nasa.gov/?v=-157.14074582245814,-38.752295066979784,2.1873791775418425,79.09951095660911&t=2018-10-18-T15%3A52%3A14Z&l=Last_of_the_Wild_1995-2004(hidden),GRUMP_Urban_Extents_Grid_1995(hidden),MODIS_Combined_L3_IGBP_Land_Cover_Type_Annual,Coastlines(hidden),Land_Water_Map)), it would be helpful to be able to toggle individual classification values (e.g., "Open Shrublands") on/off to allow the user to focus on the specific values of interest.  

This should be particularly relevant for our upcoming flood extent layer which will include a "cloud mask" value which could be distracting if a user just wants to see the observed flood areas.

**Describe the solution you'd like**
One idea is to make each "classification box" in the legend directly toggle-able.  Another would be to open the layer's setting panel and toggle it there.

**Additional context**
Eventually we'll also want the ability to toggle transparent flags for certain layers (#115).  This ticket is also remarkably similar to #232 and we'll want to talk through any differences and possibly combine them or close one as a duplicate.

### LINKED ISSUE #2571: Make image download temporary state revert actually be temporary
**Describe the bug**
When Choosing to download imagery with custom palettes, rotated imagery, or with a graticule, the imagery should only temporarily be reverted to it's original state.

**To Reproduce**
1. Go to homepage
2. Add AOD terra/modis layer
3. Change palette to a custom palette
4. Download Imagery
5. See error

**Expected behavior**
Imagery should revert to what it was before downloading imagery after finished with download

**Screenshots**
Screen Shot 2020-01-15 at 12 06 44 PM

**Additional context**
Need to also fix for GIF downloads

### PR DESCRIPTION
**Pull request by Benjaki2**
## Description

Fixes #2306
Fixes #2571 

Allow disabling of selected classifications
- [x] Add toggle in settings panel
- [x] make height of dialog based on screenHeight
- [x] Add slash through disabled legend items
- [x] create react switch component (First hook component :) )
- [x] Add temporary revert functionality  #2571 (when image download is over customs/graticules/rotations are restored)
- [x] Add toggle capability when there are dual colormaps

## Further comments

I added the toggle capability to the setting panel as opposed to a checkbox selection in the layer list because I thought it was more consistent with what we currently do. All our layer modifications are adjustable in the settings panel.

## Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

File: web/js/containers/toolbar.js [hunk 1]
```diff
@@ -239,8 +245,23 @@ function mapStateToProps(state) {
   };
 }
 const mapDispatchToProps = dispatch => ({
-  openModal: (key, customParams) => {
-    dispatch(openCustomContent(key, customParams));
+  refreshStateAfterImageDownload: (activePalettes, rotation, isGraticule) => {
+    console.log(activePalettes, rotation, isGraticule);
```
Comment (jasontk19) [inline web/js/containers/toolbar.js]: Can probably remove this console log.

File: web/js/components/layer/settings/classification-toggle.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import { CustomInput } from 'reactstrap';
```
Comment (edplato) [inline web/js/components/layer/settings/classification-toggle.js]:
```suggestion

```

File: web/js/modules/palettes/util.js [hunk 3]
```diff
@@ -338,6 +351,7 @@ export function loadPalettes(permlinkState, state) {
         var max = [];
         var squash = [];
         var count = 0;
+        console.log(layerDef);
```
Comment (edplato) [inline web/js/modules/palettes/util.js]:
```suggestion
```

**Comment by Benjaki2** [comment]
I do like how GitHub interprets the color. #theLittleThings

**Comment by Benjaki2** [comment]
Final touches
- [ ] Fix Edge colors
- [ ] Add PropTypes where they are missing
- [ ] remove logs

**Pull request merged**